rust/chains/hyperlane-cosmos/src/libs/address.rs

```diff
@@ -0,0 +1,53 @@
+use std::str::FromStr;
+
+use cosmrs::{crypto::secp256k1::SigningKey, AccountId};
+use derive_new::new;
+use hyperlane_core::{ChainResult, H256};
+
+/// decode bech32 address to H256
+pub fn bech32_decode(addr: String) -> ChainResult<H256> {
+    let account_id = AccountId::from_str(&addr)?;
+
+    // although `from_slice` can panic if the slice is not 32 bytes long,
+    // we know that we're passing in a value that is 32 bytes long because it was decoded from
```

Comment:
I believe bech32 is named that because it's the encoding is done in base 32 and not because it's 32 bytes long. I think the pattern in Cosmos is that EOA addresses are 20 bytes and smart contracts are 32 bytes. So we should probably be more defensive here

Looks like account IDs can get up to 255 bytes https://docs.rs/cosmrs/latest/cosmrs/struct.AccountId.html#associatedconstant.MAX_LENGTH

Reply:
unit tests broke this, good point